source/FreeRTOS_ND.c

```diff
@@ -1316,4 +1335,97 @@
         return xResult;
     }
 /*-----------------------------------------------------------*/
+
+/**
+ * @brief Check whether an ipv6 packet needs resolution if it is on local subnet.
+ *        If required send an ND Solicitation.
+ *
+ * @param[in] pxNetworkBuffer The network buffer with the packet to be checked.
+ *
+ * @return pdTRUE if the packet needs resolution, pdFALSE otherwise.
+ */
+    BaseType_t xCheckIPv6RequiresResolution( const NetworkBufferDescriptor_t * const pxNetworkBuffer )
+    {
+        BaseType_t xNeedsResolution = pdFALSE;
+
+        configASSERT( usFrameType( pxNetworkBuffer ) == ipIPv6_FRAME_TYPE );
+
+        /* MISRA Ref 11.3.1 [Misaligned access] */
+        /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
+        /* coverity[misra_c_2012_rule_11_3_violation] */
+        IPPacket_IPv6_t * const pxIPPacket = ( ( IPPacket_IPv6_t * ) pxNetworkBuffer->pucEthernetBuffer );
+        IPHeader_IPv6_t * const pxIPHeader = &( pxIPPacket->xIPHeader );
+        const uint8_t ucNextHeader = pxIPHeader->ucNextHeader;
+
+        if( ( ucNextHeader == ipPROTOCOL_TCP ) || ( ucNextHeader == ipPROTOCOL_UDP ) )
+        {
+            IPv6_Address_t * const pxIPAddress = &( pxIPHeader->xSourceAddress );
+            const IPv6_Type_t eType = xIPv6_GetIPType( ( const IPv6_Address_t * ) pxIPAddress );
+
```

Comment:
Yeah I can keep them there, I removed them because they were repeating the exact same info output inside 'xIPv6_GetIPType' and `eNDGetCacheEntry`. Also some of the changes I have reverted for the other PR, but haven't pushed yet.